lib/notifications/notification.ex

```diff
@@ -68,18 +63,29 @@ defmodule Notifications.Notification do
     :state
   ]
 
-  @spec get_or_create(t()) :: t()
-  def get_or_create(notification_without_id) do
+  @spec get_or_create_from_block_waiver(map()) :: t()
+  def get_or_create_from_block_waiver(block_waiver_values) do
     changeset =
-      DbNotification.changeset(%DbNotification{}, Map.from_struct(notification_without_id))
+      DbNotification.block_waiver_changeset(
+        %DbNotification{},
+        block_waiver_values
+      )
+
+    notification_values = block_waiver_values |> Map.delete(:block_id) |> Map.delete(:service_id)
 
     db_record =
       case Skate.Repo.transaction(fn ->
              with {:ok, db_record} <- insert(changeset, on_conflict: :nothing),
                   false <- is_nil(db_record.id),
-                  notification_with_id <- %__MODULE__{notification_without_id | id: db_record.id} do
+                  notification_with_id <-
+                    struct!(__MODULE__, Map.merge(notification_values, %{id: db_record.id})) do
```

Comment:
Any particular reason to use `struct!` rather than the `%__MODULE__{...}` syntax? From my quick reading of the `struct!` documentation it looks like it's equivalent.

Reply:
Hmmm, not sure if I had anything in particular in mind or if `MODULE` would work just as well...

Reply:
I remember now, there is a reason. You can't use `__MODULE__` here because it only works with literals, else you get this kind of thing:

```
** (CompileError) lib/notifications/notification.ex:81: expected key-value pairs in a map, got: Map.merge(notification_values, %{id: db_record.id})

```

Reply:
Ahh, right. And you have to use `Map.merge` rather than just `%{notification_values | id: db_record.id}` because the original map might not have an `id` key, correct?